### PR TITLE
refactor: implement Valuer/Scanner to read/write to SQL databases

### PR DESCRIPTION
**Pull request by jhoward-lm**
This PR implements the following interfaces using a simple Go template:

- `driver.Valuer` (from `database/sql/driver`)
- `sql.Scanner` (from `database/sql`)

This is to enable storing protobom wire format messages as raw bytes in a SQL database, needed by protobom/storage#29 as a more efficient storage format.

More info [here](https://entgo.io/docs/faq#how-to-store-protobuf-objects-in-a-blob-column).

### Other changes

- bump `protoc` version to 28.2
- bump `protoc-gen-go` version to v1.34.2
- bump `golangci-lint` version in workflow to v1.61, address new findings

### CODE REVIEW

File: api/template/value_scanner.go.tmpl [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	return value(x)
+}
+
+func (x *{{ $name }}) Scan(src any) error {
+	return scan(src, x)
+}
+{{- end }}
+
+func value(msg proto.Message) (driver.Value, error) {
+	return proto.Marshal(msg)
```
Comment (jhoward-lm) [inline api/template/value_scanner.go.tmpl:24]: Change to deterministic

**Review by cpanato — DISMISSED** [review]
lgtm

thanks

**Comment by jhoward-lm** [comment]
@cpanato had to push a small change, mind approving again?

**Review by cpanato — APPROVED** [review]

File: api/render.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+//go:build ignore
```
Comment (puerco) [inline api/render.go:1]: nit: add the copyright boilerplate (and to generate.go)

**Review by puerco — COMMENTED** [review]
Awesome, lgtm. One small nit about the copyright headers but looks good to go :rocket:

**Review by puerco — APPROVED** [review]

File: api/render.go (hunk 2 above)
Comment (puerco) [reply api/render.go:6]: Thanks! Ship it! :shipit:

Comment (jhoward-lm) [reply api/render.go:6]: Thanks! But I can't merge. Mind doing it? And would you be ok with cutting a patch release with it?

**Pull request merged**